JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/User.tests.ts

```diff
@@ -105,6 +109,295 @@ class UserContextTests extends TestClass {
+                Assert.equal(undefined, user.accountId, "user account id was not set");
+                Assert.equal(cookieStub.notCalled, true, "cookie was not set");
+                Assert.equal(loggingStub.calledOnce, true, "Warning was logged");
+                cookieStub.restore();
+                loggingStub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "setAuthenticatedUserContext: handles authId special characters correctly",
+            test: () => {
+                // setup
+                var authAndAccountId = ['my|||special;id', '1234'];
+                var user = new Microsoft.ApplicationInsights.Context.User(undefined);
+                var cookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "setCookie");
+                var loggingStub = sinon.stub(Microsoft.ApplicationInsights._InternalLogging, "throwInternalUserActionable");
+
+                // act
+                user.setAuthenticatedUserContext(authAndAccountId[0], authAndAccountId[1]);
+
+                // verify
+                Assert.equal(undefined, user.authenticatedId, "user auth id was not set");
+                Assert.equal(undefined, user.accountId, "user account id was not set");
+                Assert.equal(cookieStub.notCalled, true, "cookie was not set");
+                Assert.equal(loggingStub.calledOnce, true, "Warning was logged");
+                cookieStub.restore();
+                loggingStub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "setAuthenticatedUserContext: handles accountId special characters correctly",
+            test: () => {
+                // setup
+                var authAndAccountId = ['myid', '1234 5678'];
+                var user = new Microsoft.ApplicationInsights.Context.User(undefined);
+                user.clearAuthenticatedUserContext();
+                var cookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "setCookie");
+                var loggingStub = sinon.stub(Microsoft.ApplicationInsights._InternalLogging, "throwInternalUserActionable");
+
+                // act
+                user.setAuthenticatedUserContext(authAndAccountId[0], authAndAccountId[1]);
+
+                // verify
+                Assert.equal(undefined, user.authenticatedId, "user auth id was not set");
+                Assert.equal(undefined, user.accountId, "user account id was not set");
+                Assert.equal(cookieStub.notCalled, true, "cookie was not set");
+                Assert.equal(loggingStub.calledOnce, true, "Warning was logged");
+                cookieStub.restore();
+                loggingStub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "setAuthenticatedUserContext: handles unicode characters correctly",
+            test: () => {
+                // setup
+                var authAndAccountId = ['שלום', 'להתראות'];
```

Comment:
here and in other places - for unicode specific tests, could you use '\xxx' syntax?